### ISSUE TITLE
loss doesn't drop when setting the **groups** of the convolutional layer to **in_channel**

### ISSUE THREAD

**Issue by EricAllen16** [jiecaoyu/XNOR-Net-PyTorch#85]
First, thank you for sharing the great work! Now I have a problem to ask you for help. When I borrow the design idea of Depthwise separable CNN and set the _**groups**_ of the convolutional layer to _**in_channel**_ in the Depthwise Conv stage, the loss doesn't drop. What is the reason for it? Thanks.

**Comment by jiecaoyu** [comment]
Hi @EricAllen16 , could you provide more details (e.g., which line and what change)?

**Comment by EricAllen16** [comment]
Thank you for your quick reply! Actually, I change the network structure for my own dataset training. And when I set the **_groups_** of the convolutional layer to 1, it runs fine. But if I set the **_groups_** of the convolutional layer to _**in_channel**_ in the Depthwise Conv stage like the following picture, the loss doesn’t drop. Could this be related to the “meancenterConvParams” function? Because the dimension of weights' tensor would be (out_channel, 1, kernel_size[0], kernel_size[1]) when setting the _**groups**_ of the convolutional layer to _**in_channel**_. Or something else cause it? I have no idea.
![微信图片_20200325161402](https://user-images.githubusercontent.com/56765142/77515564-b13d3000-6eb3-11ea-90ef-3684cc38e1d1.png)

**Comment by jiecaoyu** [comment]
Okay, do you mean that your training process does not converge? I think completely replacing the conventional convolutional layers with the depth-wise convolutional layers is too aggressive. It is better to first test whether the all-floating-point version converges or not.

**Comment by EricAllen16** [comment]
Yes, I have tested the all-floating-point version, it runs fine. I also tested the binarization of other convolutional layers except the depth-wise convolutional layer and it can also converge. But when I binarize the depth-wise convolutional layers, the training process does not converge.

**Comment by jiecaoyu** [comment]
@EricAllen16 I think binarizing depth-wise convolution is too aggressive, and it may need higher precision or better quantization method.

**Comment by EricAllen16** [comment]
Thank you for your advice!
